Wordle.Api/Wordle.Api/Controllers/TokenController.cs

```diff
@@ -0,0 +1,101 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Wordle.Api.Dtos;
+using Wordle.Api.Identity;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class TokenController : ControllerBase
+    {
+        public WordleDbContext _context;
+        public UserManager<AppUser> _userManager;
+        public JwtConfiguration _jwtConfiguration;
+        public RoleManager<IdentityRole> _roleManager;
+        public TokenController(WordleDbContext context, UserManager<AppUser> userManager, JwtConfiguration jwtConfiguration, RoleManager<IdentityRole> roleManager)
+        {
+            _context = context;
+            _userManager = userManager;
+            _jwtConfiguration = jwtConfiguration;
+            _roleManager = roleManager;
+        }
+
+        [HttpPost("GetToken")]
+        public async Task<IActionResult> GetToken([FromBody] UserCredentialsDto userCredentials)
+        {
+            if (string.IsNullOrEmpty(userCredentials.Username))
+            {
+                return BadRequest("Username is required");
+            }
+            if (string.IsNullOrEmpty(userCredentials.Password))
+            {
+                return BadRequest("Password is required");
+            }
+
+            var user = _context.Users.FirstOrDefault(u => u.UserName == userCredentials.Username);
+
+            if (user is null) { return Unauthorized("The user account was not found"); }
+
+            // Verify the user's password - Microsoft will do this for us. 
+            bool results = await _userManager.CheckPasswordAsync(user, userCredentials.Password);
+            if (results)
+            {
+                // Set up our security key and our hashing algorithm
+                // PS we are storing our key in a pretty unsafe area... Do not store your JWT config key as plain text like this in production level applications\
+                // If someone gets this token secret they can generate tokens for themselves
+                var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtConfiguration.Secret));
+                var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+                var claims = new List<Claim>
+                {
+                    new(JwtRegisteredClaimNames.Sub, user.UserName!),
+                    new(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+                    new("userId", user.Id.ToString()),
+                    new("userName", user.UserName!.ToString().Substring(0,user.UserName.ToString().IndexOf("@"))), // Use the email as the username, but get rid of the email domain
+                    new(Claims.Random, (new Random()).NextDouble().ToString()),
+                    new(Claims.BirthDate, user.BirthDate.ToString()),
+                    new(Claims.MasterOfTheUniverse, user.IsMasterOfTheUniverse.ToString())
```

Comment:
If you add this claim on user creation (identitySeed), then this line would not be necessary

Reply:
Got it. Thanks :)